server/migrations/20240212232936-modify-collectionjob-id-type.js

```diff
@@ -0,0 +1,63 @@
+'use strict';
+
+module.exports = {
+    up: async queryInterface => {
+        // Drop the primary key constraint to enable changing the 'id' column type
+        await queryInterface.sequelize.query(
+            `ALTER TABLE "CollectionJob" DROP CONSTRAINT "CollectionJob_pkey";`
+        );
+
+        // Change the 'id' column type to INTEGER
+        await queryInterface.sequelize.query(
+            `ALTER TABLE "CollectionJob" ALTER COLUMN "id" TYPE INTEGER USING ("id"::integer);`
+        );
+
+        // Create a sequence for auto-increment
+        await queryInterface.sequelize.query(
+            `CREATE SEQUENCE collectionjob_id_seq AS INTEGER;`
+        );
+
+        // Set the default value of 'id' to the next value of the sequence
+        await queryInterface.sequelize.query(
+            `ALTER TABLE "CollectionJob" ALTER COLUMN "id" SET DEFAULT nextval('collectionjob_id_seq');`
+        );
+
+        // Set the sequence as the owner of the 'id' column
+        await queryInterface.sequelize.query(
+            `ALTER SEQUENCE collectionjob_id_seq OWNED BY "CollectionJob"."id";`
+        );
+
+        // Ensure that the sequence starts from the next value of the current maximum 'id'
+        await queryInterface.sequelize.query(`
+            SELECT setval(pg_get_serial_sequence('"CollectionJob"', 'id'), 
+            COALESCE((SELECT MAX("id") FROM "CollectionJob"), 0) + 1);
+        `);
+
+        // Add the primary key constraint back
+        await queryInterface.sequelize.query(
+            `ALTER TABLE "CollectionJob" ADD PRIMARY KEY ("id");`
+        );
+    },
+
+    down: async queryInterface => {
+        // Remove the primary key constraint
+        await queryInterface.sequelize.query(
+            `ALTER TABLE "CollectionJob" DROP CONSTRAINT "CollectionJob_pkey";`
+        );
+
+        // Drop the sequence created for auto-increment
+        await queryInterface.sequelize.query(
+            `DROP SEQUENCE collectionjob_id_seq;`
+        );
```

Comment:
This fails on my system:

>     $ yarn sequelize db:migrate:undo
>     yarn run v1.22.19
>     $ dotenv -e config/dev.env npx sequelize-cli db:migrate:undo
>     
>     Sequelize CLI [Node: 18.18.0, CLI: 6.6.0, ORM: 6.28.0]
>     
>     Loaded configuration file "config/config.js".
>     Using environment "development".
>     == 20240212232936-modify-collectionjob-id-type: reverting =======
>     
>     ERROR: cannot drop sequence collectionjob_id_seq because other objects depend on it
>     ERROR DETAIL: default value for column id of table "CollectionJob" depends on sequence collectionjob_id_seq
>     
>     error Command failed with exit code 1.
>     info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

Reply:
Bummer! I'll check it out when I get more time back on the project

Reply:
This should be fixed with [5769bd3](https://github.com/w3c/aria-at-app/pull/925/commits/5769bd3e06bfbece9e83c183f8b9382a73f415b5). I tested both up and down with default test data and a dump of prod's DB from today.